torch/_subclasses/fake_tensor.py

```diff
@@ -567,6 +567,14 @@ def merge_devices(t):
         tree_map(merge_devices, args)
         tree_map(merge_devices, kwargs)
 
+        # some functions that allow Python numbers to bind to Tensors
+        # if we have failed to find a device, and we're running one of these operators,
+        # we must have scalar only inputs
+        if torch._C._should_allow_numbers_as_tensors(
+            func.name().split("::")[-1].split(".")[0]
```

Comment:
I feel that func already has an API for getting this directly, and if it does not, you should add one.

Reply:
`unqualified_name` I guess ?